src/exec/mod.rs

```diff
@@ -176,10 +180,22 @@ impl CommandTemplate {
         for arg in &self.args[1..] {
             if arg.has_tokens() {
                 // A single `Tokens` is expected
-                // So we can directy consume the iterator once and for all
-                for path in &mut paths {
-                    cmd.arg(arg.generate(&path).as_ref());
-                    has_path = true;
+                // So we can directly consume the iterator once and for all
+
+                // It would be too expansive to collect all the paths at once, so we take
+                // only the first ones.
+                let mut first_paths = take(&mut paths, SORT_THRESHOLD);
+                has_path = !first_paths.is_empty();
+
+                if first_paths.len() < SORT_THRESHOLD {
```

Comment:
shouldn't this be `<= SORT_THRESHOLD`?

Reply:
Actually, as it is now `first_paths.len() <= SORT_THRESHOLD` is always true. Maybe it's clearer if we refactor as
```rust
let mut first_paths = take(&mut paths, SORT_THRESHOLD + 1);

if first_paths.len() <= SORT_THRESHOLD {
```

Reply:
I'm not sure I understand what the intention is?

Reply:
I appreciate that you were concerned about performance and introduced the `SORT_THRESHOLD`, but I am not sure if this is really required. I would expect the sort time to be much smaller than the actual filesystem-search. I would therefore remove the limit and simplify the logic here. There is a limit to the number of paths for `-X` anyway (see #410), so I don't think this will cause any problems.

Reply:
Our motivating reason was [this](https://github.com/sharkdp/fd/issues/441#issuecomment-499675495). If sorting time is not an issue, please close this pull request and we'll submit another with less noise.

Reply:
Ah, sorry. I forgot about my *"If we would not be restricted, it might make sense to only sort the results if there are less than some fixed number."* comment.

I would like to start with an easy solution where we always sort the paths. If this turns out to be a problem (performance wise), we can go back to an approach like this.